content/contributing.md

```diff
@@ -1,3 +1,105 @@
+
+> NOTE: The CI job will check the formatting and local links, so before committing, make sure to run `make docs` from the main repo root!
+
+1. Create PR
+
+2. Once the PR is merged, the content will appear on our website `https://rhobs-handbook.netlify.app/` and [GitHub UI](https://github.com/rhobs/handbook/tree/main/content).
+
+#### Adding Static Content
+
+All static content that you want to link to or embed (e.g. images) add to the `content` (ideally to [`content/assets`](https://github.com/rhobs/handbook/tree/main/content/assets) for clean grouping) and relatively link in your main document.
+
+It's recommended to check if your page renders well before merging (see [section below](#early-previews)).
+
+> NOTE: If you used some tool for creating diagrams (e.g checkout amazing https://excalidraw.com/), put the source file into assets too. If online tool was used, e.g Google Draw, add link to the markdown so we can edit those easily.
+
+#### Early Previews
+
+While editing documentation, you can run the website locally to check how it all looks!
+
+*Locally:*
+
+Run `make web-serve` from repo root. This will serve the website on `localhost:1313`.
+
+> NOTE: Currently some options reloads in runtime e.g CSS, but content changes requires `make web-serve` restart. See https://github.com/bwplotka/mdox/issues/35 to track fixing this.
+
+*From the PR:*
+
+Click `details` on website preview CI check from netlify:
+
+![preview](assets/screen-preview.png)
+
+> Something does not work? Link works on GitHub but not on the website? Don't try to hack this up. It might bug with our framework. Please raise an issue on https://github.com/rhobs/handbook/ for us to investigate.
+
+## Advanced: Under the hood
+
+We do advanced conversions from source files in `content` dir to the website. Let's do a quick overview of our framework.
+
+* [`./web`](https://github.com/rhobs/handbook/tree/main/web) contains resources related to the website.
+* [`./web/themes`](https://github.com/rhobs/handbook/tree/main/web/themes) hosts our Hugo theme [`Docsy`](https://themes.gohugo.io/docsy/) as git submodule.
+* [`./web/config.toml`](https://github.com/rhobs/handbook/blob/main/web/config.toml) holds Hugo and theme configuration.
+* [`./netlify.yaml`](https://github.com/rhobs/handbook/blob/main/netlify.toml) holds our Netlify deployment configuration.)
+* [`.mdox.yaml`](https://github.com/rhobs/handbook/blob/main/.mdox.yaml) holds https://github.com/bwplotka/mdox configuration that tells mdox how to *pre-process* our docs.
+
+When you run `make web` or `make web-server` a particular process is happening to "render" the website:
+
+1. `mdox` using `.mdox.yaml` is transforming the resources from `content` directory, moving into website friendly form (adding frontmatter, adjusting links and many others) and outputs results in `./web/content` and `./web/static` which are never committed (temporary directories).
```

Comment:
Is there an actual need for `mdox`? Could we simply stick with mdbook, mkdocs or some other widely used tool for static page generation?

Reply:
I kind of set ambitous goal to make sure you don't worry about website - you just write markdown like you would want to have it exposed on GH UI, mdox will do rest for you. We are growing this project as default website preprocess/tooling with @saswatamcode 

If this become a bottleneck we can switch to just hugo with magic fontmatters and directives, but let's see what the experience is maybe?

Reply:
Ok, let's give it a shot :+1: